pkg/analyze/deployment_status.go

```diff
@@ -24,7 +24,7 @@ func analyzeDeploymentStatus(analyzer *troubleshootv1beta2.DeploymentStatus, get
 	var status *appsv1.DeploymentStatus
 	for _, deployment := range deployments {
 		if deployment.Name == analyzer.Name {
-			status = &deployment.Status
```

Comment:
@codelingobot Prevent reference assignment of local variable in a loop to a variable used outside of the loop